libwallet/assets/btc/sync.go

```diff
@@ -191,26 +191,30 @@ func (asset *Asset) updateSyncProgress(rawBlockHeight int32) {
 	asset.syncData.headersFetchProgress.HeadersFetchProgress = int32((headersFetchedSoFar * 100) / allHeadersToFetch)
 	asset.syncData.headersFetchProgress.GeneralSyncProgress.TotalSyncProgress = asset.syncData.headersFetchProgress.HeadersFetchProgress
 	asset.syncData.headersFetchProgress.GeneralSyncProgress.TotalTimeRemainingSeconds = int64((timeSpentSoFar * remainingHeaders) / headersFetchedSoFar)
+	asset.syncData.mu.Unlock()
 
 	// publish the sync progress results to all listeners.
+	asset.syncData.mu.RLock()
 	for _, listener := range asset.syncData.syncProgressListeners {
 		if listener.OnHeadersFetchProgress != nil {
 			listener.OnHeadersFetchProgress(&asset.syncData.headersFetchProgress)
 		}
 	}
+	asset.syncData.mu.RUnlock()
```

Comment:
This is the problem area causing the freezes. updateSyncProgress in holds the syncData lock until its finished. At then end it does listener.OnHeadersFetchProgress which the info page does a updateSyncProgress which does pg.fetchSyncProgress which does pg.deleteSyncProgress which tries wal.IsSynced which needs a syncData read lock that it cannot get.